anchore_engine/analyzers/syft/__init__.py

```diff
@@ -1,12 +1,39 @@
 import collections
 
-from anchore_engine.analyzers.utils import defaultdict_to_dict, content_hints
+from anchore_engine.analyzers.utils import defaultdict_to_dict, content_hints, dig
 from anchore_engine.clients.syft_wrapper import run_syft
 from .handlers import modules_by_artifact_type, modules_by_engine_type
 
 
-def filter_artifacts(artifact):
-    return artifact["type"] in modules_by_artifact_type
+def filter_relationships(relationships, **kwargs):
+    def filter_fn(relationship):
+        for key, expected in kwargs.items():
+            if relationship[key] != expected:
+                return False
+        return True
+
+    return filter(filter_fn, relationships)
+
+
+def filter_artifacts(artifacts, relationships):
+    def filter_fn(artifact):
+        # syft may do more work than what is supported in engine, ensure we only include artifacts
+        # of select package types.
+        if artifact["type"] not in modules_by_artifact_type:
+            return False
+
+        # some packages are owned by other packages (e.g. a python package that was installed
+        # from an RPM instead of with pip), filter out any packages that are not "root" packages.
+        if list(
```

Comment:
if this is casting to a `list` because `filter` is returning a generator, then I don't see why using `filter()` is useful in the first place

Reply:
the condition is testing truthiness, the generator will always be truthy, which is why the we pass the generator to the list constructor to take the generator and evaluate the filter (apply the filter to each item to obtain a new list).

I could swap the return of `filter_artifacts` and `filter_relationships` to be a list and not use `filter`? Something like `return [x for x in artifacts if filter_fn(x)]` @alfredodeza

Reply:
Right, I get that most all data structures in Python evaluate as truthy - my suggestion was that it seems that there is no use for the generator since this is forcing it to be a list. Instead make it a list from the beginning. 

The list could be a list comprehension like you suggested, seems reasonable